File: server/Processors/SteamRemoteStorage.cs
using Airmiss.Processor;
using System.Text;

[ProcessorHub("remotestorage")]
public class SteamRemoteStorage
{
    [Processor("enabled", Verb.Get)]
    public bool IsCloudEnabled()
    {
        return Steamworks.SteamRemoteStorage.IsCloudEnabled;
    }

    [Processor("{filename}", Verb.Delete)]
    public bool DeleteFile([Path("filename")] string filename)
    {
        return Steamworks.SteamRemoteStorage.FileDelete(filename);
    }
    
    [Processor("{filename}/exists", Verb.Get)]
    public bool HasFile([Path("filename")] string filename)
    {
        return Steamworks.SteamRemoteStorage.FileExists(filename);
    }

    [Processor("{filename}", Verb.Get)]
    public string ReadFile([Path("filename")] string filename)
    {
        return Encoding.UTF8.GetString(Steamworks.SteamRemoteStorage.FileRead(filename));
    }

    [Processor("{filename}", Verb.Post)]
    public bool WriteFile([Path("filename")] string filename, [Content] string content)
    {
        return Steamworks.SteamRemoteStorage.FileWrite(filename, Encoding.UTF8.GetBytes(content));
    }
}

File: client/SteamRemoteStorage.ts
import { Api, Verb } from './Api.js'

export class SteamRemoteStorage {
    public static async isCloudEnabled(): Promise<boolean> {
        return await Api.send('/remotestorage/enabled', Verb.Get)
    }

    public static async deleteFile(name: string): Promise<boolean> {
        return await Api.send(`/remotestorage/${name}`, Verb.Delete)
    }

    public static async hasFile(name: string): Promise<boolean> {
        return await Api.send(`/remotestorage/${name}/exists`, Verb.Get)
    }

    public static async readFile(name: string): Promise<boolean> {
        return await Api.send(`/remotestorage/${name}`, Verb.Get)
    }

    public static async writeFile(name: string, content: string): Promise<boolean> {
        return await Api.send(`/remotestorage/${name}`, Verb.Post, content)
    }
}
